contracts/proxy/redeem-queue/contract.rs

```diff
@@ -54,42 +60,211 @@ fn vault_reserve_balance(deps: Deps, hub: &str, vault: &str) -> Result<u128, Err
     Ok(vault_metadata.reserve_balance.u128())
 }
 
-pub fn handle_process_head(_deps: DepsMut, _vault: String) -> Result<Response, Error> {
-    todo!()
+pub fn handle_process_head(deps: DepsMut, vault: String) -> Result<Response, Error> {
+    deps.api.addr_validate(&vault)?;
+
+    let hub = deps.storage.hub();
+    let reserve_balance = vault_reserve_balance(deps.as_ref(), &hub, &vault)?;
+    let synthetic = synthetic_for_vault(deps.as_ref(), &hub, &vault)?;
+
+    let mut queue = RedemptionQueue::new(deps.storage, &vault);
+    let available_amount = Uint128::new(reserve_balance);
```

Comment:
We could reduce this to a single query:

```suggestion
    let vault_metadata: VaultMetadata = deps.querier.query_wasm_smart(
        &hub,
        &HubQueryMsg::VaultMetadata {
            vault: vault.to_owned(),
        },
    )?;
    let synthetic = vault_metadata.synthetic;
    let available_amount = vault_metadata.reserve_balance;
    
    let mut queue = RedemptionQueue::new(deps.storage, &vault);
```